lib/util.c

```diff
@@ -2193,3 +2193,137 @@ EXPORTED char *modseqtoa(modseq_t modseq)
     buf_printf(&buf, MODSEQ_FMT, modseq);
     return buf_release(&buf);
 }
+
+static char *_xsyslog_ev_escape(const char *val)
+{
+    struct buf buf = BUF_INITIALIZER;
+    int needs_escaping = 0, orig_len, escaped_len;
+    const char *p;
+
+    buf_setcstr(&buf, val);
+
+    escaped_len = orig_len = buf_len(&buf);
```

Comment:
buf_len returns a size_t so why cast implicitly to int here?
nit: from C99 onwards there's no need to declare variables at the start of the function.

Reply:
Fixed the size_t issue. I think there's still a habit of initialize at the top we haven't fully broken yet :) I've left that bit...

Reply:
Collecting variable declarations at the top of their scope is a good habit in languages with manual memory management.  We don't have to copy C++'s bad habits just because we now can.

Reply:
> Collecting variable declarations at the top of their scope is a good habit in languages with manual memory management. We don't have to copy C++'s bad habits just because we now can.

I disagree. But there is no point in back and forth without an agreed code style for this project, so I'll resolve this comment.

Reply:
Ah, gcc does when -O is involved. Wild.